core/testsuite/src/test/java/com/blazebit/persistence/testsuite/Issue571Test.java

```diff
@@ -0,0 +1,342 @@
+                CriteriaBuilder<MyEntity> select = cbf.create(em, MyEntity.class)
+                        .withRecursive(Cte.class)
+                        .from(MyEntity.class, "a")
+                        .bind("myEntity").select("a")
+                        .bind("id").select("1")
+                        .unionAll()
+                        .from(MyEntity.class, "a")
+                        .where("a.attribute").eq("bogus")
+                        .bind("myEntity").select("a")
+                        .bind("id").select("1")
+                        .end()
+                        .from(Cte.class)
+                        .select("myEntity");
+
+                List<MyEntity> resultList = select
+                        .getResultList();
+
+                String sql = ((CustomSQLTypedQuery) select.getQuery()).getQuerySpecification().getSql();
+                String cteBase = sql.substring(sql.indexOf("\nselect") + 1, sql.lastIndexOf("UNION ALL") - 1);
+                String cteRecursivePart = sql.substring(sql.indexOf("UNION ALL")+10, sql.lastIndexOf("select")-3);
+                String queryPart = sql.substring(sql.lastIndexOf("select"));
+
+                assertEquals("Recursive base part should project 2 columns separated by a single comma",
+                        cteBase.indexOf(","), cteBase.lastIndexOf(","));
+
+                assertEquals("Recursive part should project 2 columns separated by a single comma",
+                        cteRecursivePart.indexOf(","), cteRecursivePart.lastIndexOf(","));
+
+                assertEquals("Final query should project 2 columns separated by a single comma",
+                        queryPart.indexOf(","), queryPart.lastIndexOf(","));
+            }
+        });
+    }
+
+    @Test
+    @Ignore("It should not be possible to bind an association to its identifier, instead it should be bound to its association type")
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQL.class, NoOracle.class, NoHibernate42.class, NoHibernate43.class  })
+    public void testBindingCteAssociationToEntityId() {
+        transactional(new TxVoidWork() {
+            @Override
+            public void work(EntityManager em) {
+                CriteriaBuilder<MyEntity> select = cbf.create(em, MyEntity.class)
+                        .withRecursive(Cte.class)
+                        .from(MyEntity.class, "a")
+                        .bind("myEntity").select("a.id")
+                        .bind("id").select("1")
+                        .unionAll()
+                        .from(MyEntity.class, "a")
+                        .where("a.attribute").eq("bogus")
+                        .bind("myEntity").select("a.id")
+                        .bind("id").select("1")
+                        .end()
+                        .from(Cte.class)
+                        .select("myEntity");
+
+                List<MyEntity> resultList = select
```

Comment:
Instead of ignoring the test, please catch the exception and verify that the right exception is thrown i.e. check for a known part of the exception message.

Reply:
Sure, just wasn't sure if we wanted the exception in the first place (i.e. because we don't exactly know how to properly detect it